x/nexus/keeper/address.go

```diff
@@ -9,20 +9,33 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-func (k Keeper) setLatestDepositAddress(ctx sdk.Context, depositChain string, recipientAddress, depositAddress exported.CrossChainAddress) {
-	k.getStore(ctx).Set(latestDepositAddressPrefix.AppendStr(depositChain).Append(utils.LowerCaseKey(recipientAddress.String())), &depositAddress)
+func getLatestDepositAddressKey(depositChain string, recipientAddress exported.CrossChainAddress) utils.Key {
+	return latestDepositAddressPrefix.
+		Append(utils.LowerCaseKey(depositChain)).
+		Append(utils.LowerCaseKey(recipientAddress.Chain.Name)).
+		Append(utils.KeyFromStr(recipientAddress.Address))
+}
+
+func (k Keeper) setLatestDepositAddress(ctx sdk.Context, recipientAddress, depositAddress exported.CrossChainAddress) {
+	k.getStore(ctx).Set(getLatestDepositAddressKey(depositAddress.Chain.Name, recipientAddress), &depositAddress)
 }
 
 func (k Keeper) getLatestDepositAddress(ctx sdk.Context, depositChain string, recipientAddress exported.CrossChainAddress) (depositAddress exported.CrossChainAddress, ok bool) {
-	return depositAddress, k.getStore(ctx).Get(latestDepositAddressPrefix.AppendStr(depositChain).Append(utils.LowerCaseKey(recipientAddress.String())), &depositAddress)
+	return depositAddress, k.getStore(ctx).Get(getLatestDepositAddressKey(depositChain, recipientAddress), &depositAddress)
+}
+
+func getLinkedAddressesKey(depositAddress exported.CrossChainAddress) utils.Key {
+	return linkedAddressesPrefix.
+		Append(utils.LowerCaseKey(depositAddress.Chain.Name)).
+		Append(utils.KeyFromStr(depositAddress.Address))
```

Comment:
```suggestion
		Append(utils.LowerCaseKey(depositAddress.Address))
```
Just in case if the deposit address is passed in as a mixed case.